core/kernel/asan.c

```diff
@@ -411,11 +411,18 @@ void __asan_handle_no_return(void)
 void __asan_register_globals(struct asan_global *globals, size_t size);
 void __asan_register_globals(struct asan_global *globals, size_t size)
 {
-	size_t n;
+	size_t n = 0;
+
+	for (; n < size; n++) {
+		vaddr_t end = globals[n].beg + globals[n].size;
+		vaddr_t end_align = ROUNDUP(end, ASAN_BLOCK_SIZE);
+		vaddr_t end_rz = globals[n].beg
+			+ globals[n].size_with_redzone;
+		vaddr_t begin = globals[n].beg;
```

Comment:
Since you've added this helper variable, why don't you move it up a bit so it can be used in the other expressions?

Reply:
Done